client/my-sites/plugins/plugin-management-v2/update-plugin/index.tsx

```diff
@@ -53,63 +53,46 @@ export default function UpdatePlugin( {
 			( selectedSite ? parseInt( status.siteId ) === selectedSite.ID : true )
 	);
 
-	const handleUpdateConfirmation = () => {
-		setDisplayConfirmModal( true );
-	};
-
 	const pluginUpdateConfirmationTitle = translate( 'Update %(plugin)s', {
 		args: {
 			plugin: plugin.name ?? plugin.slug,
 		},
 	} );
 
-	const showConfirmationModal = () => (
-		<ConfirmModal
-			isVisible={ displayConfirmModal }
-			confirmButtonLabel={ translate( 'Update' ) }
-			text={ translate(
-				'You are about to update the %(plugin)s plugin to version %(version)s, on %(siteCount)d site. ',
-				'You are about to update the %(plugin)s plugin to version %(version)s, on %(siteCount)d sites. ',
-				{
-					count: siteCount ?? 1,
-					args: {
-						version: updatedVersions[ 0 ],
-						plugin: plugin.name ?? plugin.slug,
-						siteCount: String( siteCount ),
-					},
-				}
-			) }
-			title={ String( pluginUpdateConfirmationTitle ) }
-			onCancel={ () => setDisplayConfirmModal( false ) }
-			onConfirm={ () => {
-				updatePlugin && updatePlugin( plugin );
-			} }
-		/>
-	);
+	const onUpdatePlugin = useCallback( () => {
+		updatePlugin && updatePlugin( plugin );
+		setDisplayConfirmModal( false );
```

Comment:
This is the main fix. The rest of the codes are just improvements for better code readability.